snuba/clickhouse/columns.py

```diff
@@ -429,23 +439,23 @@ def _repr_content(self) -> str:
     def __eq__(self, other: object) -> bool:
         return (
             self.__class__ == other.__class__
-            and self.get_modifiers() == cast(Enum, other).get_modifiers()
-            and self.values == cast(Enum, other).values
+            and self.get_modifiers() == cast(Enum[TModifiers], other).get_modifiers()
+            and self.values == cast(Enum[TModifiers], other).values
         )
 
     def _for_schema_impl(self) -> str:
         return "Enum({})".format(
             ", ".join("'{}' = {}".format(v[0], v[1]) for v in self.values)
         )
 
-    def set_modifiers(self, modifiers: Optional[TypeModifiers]) -> Enum:
+    def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
         return Enum(values=self.values, modifiers=modifiers)
 
-    def get_raw(self) -> Enum:
+    def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
 
 
-class ColumnSet:
+class ColumnSet(Generic[TModifiers]):
```

Comment:
This class doesn't need to be generic since it's only used in schemas and not migrations

Reply:
you are right, I did not notice that. Let me remove this.

Reply:
thanks